### PR TITLE
ocaml: new packages (II)

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Almost done.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by pmetzger** [comment]
This is really going to be a nightmare to maintain going forward, fwiw. We're basically duplicating the work of opam.

File: ocaml/ocaml-menhir/Portfile [hunk 1]
```diff
@@ -4,22 +4,21 @@ PortSystem          1.0
...
 name                ocaml-menhir
-version             20210310
+version             20230608
 revision            0
 categories          ocaml devel
-platforms           darwin
-maintainers         nomaintainer
-license             {QPL LGPL}
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+license             GPL-2
```
Comment (pmetzger) [inline ocaml/ocaml-menhir/Portfile:11]: Menhir is LGPL-2. It is not GPL-2.

**Review by pmetzger — CHANGES_REQUESTED** [review]

File: ocaml/ocaml-menhir/Portfile (hunk 1 above)
Comment (pmetzger) [reply ocaml/ocaml-menhir/Portfile:11]: Actually I'm mistaken. Part of it is LGPL (the runtime), part is GPL-2 (the generator.)

Comment (pmetzger) [reply ocaml/ocaml-menhir/Portfile:11]: https://gitlab.inria.fr/fpottier/menhir/-/blob/master/LICENSE

**Comment by barracuda156** [comment]
@pmetzger I am not gonna pull in many more; my aim here is just to support Stan.

To that point we do need a few fixes which are lacking elsewhere (yes, I know older systems is not your concern, but I wanna have Stan on PPC).

Basically, with this PR we have everything to build Stan compiler. Test dependencies will need 2–3 extra ports perhaps, but that is it.

This is not R case, where I wanted to reimplement the whole ecosystem :) Once we have Stan, I am done with OCaml packages.

P. S. I thought to add one port which depends on Coq, for Category Theory. It if works, it won’t require another 70 ports to support it.

File: ocaml/ocaml-menhir/Portfile (hunk 1 above)
Comment (barracuda156) [reply ocaml/ocaml-menhir/Portfile:11]: Thank you, I will fix this in half an hour.

Comment (barracuda156) [reply ocaml/ocaml-menhir/Portfile]: @pmetzger Fixed, rebased.

**Pull request merged**